docs/general/attributes.md

```diff
@@ -306,12 +306,13 @@ Examples of `peer.service` that users may specify:
 
 ## General identity attributes
 
-These attributes may be used for any operation with an authenticated and/or authorized enduser.
+These attributes may be used for any operation with an authenticated or anonymous enduser.
 
 <!-- semconv identity -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
 | `enduser.id` | string | Username or client_id extracted from the access token or [Authorization](https://tools.ietf.org/html/rfc7235#section-4.2) header in the inbound request from outside the system. | `username` | Recommended |
+| `enduser.anon_id` | string | Randomly generated id of an anonymous user that is persisted throughout the lifetime of the visits. | `V1StGXR8_Z5jdHi6B-myT` | Recommended |
```

Comment:
@LikeTheSalad  The same browser instance can be used to have multiple sessions at different times but the anonymous user will be the same in all these sessions.

Reply:
Got it, it's similar to [device.id](https://github.com/open-telemetry/semantic-conventions/blob/de4125d0307e97474ea05d40b3c7945a027745b3/docs/resource/device.md?plain=1#L12) ?

Reply:
yes, it's close to device.id, but not the same. Browsers, including on mobile, in incognito mode will generate a new `enduser.anon_id`. For mobile apps is there a possibility of ever requiring a different anon_id? If not, then I guess `device.id` can be used for `enduser.anon_id`.

Reply:
Mildly 🌶 take but I would say people used to web dev would be understanding that incognito and cross-TLD (considering cookie limitations) would lead to multiple IDs from same "physical" device - even if the attribute would be called `device.id`

So I'd be fine with just using `device.id` for this use case with asterisks that "runtime sandboxes" could lead to multiple different ID's for physical device

Reply:
@t2t2 I am concerned about the intent of this attribute. The reason to put this under `enduser` namespace is really to represent an anonymous user, although its value can be device's id whenever it's available. And this attribute goes beyond Client side situations - for anonymous sessions, the backend telemetry too can include the anonymous user id too - the clients will have to pass it via baggage or some such mechanism to the backends.  In this context, I also want to note that we cannot use `session.id` for this purpose since the same anonymous user can have multiple sessions.

I still feel it's good to have this attribute `enduser.anon_id` to represent anonymous users.

@open-telemetry/specs-semconv-approvers I am curious what your thoughts are on this since this goes beyond Client instrumentation concerns.

Reply:
@open-telemetry/specs-semconv-approvers  can we get some attention on this please? This is an attribute that some of us vendors would like to add to all signals via processors, and so shouldn't be an issue to others. Is there a general guidance on semantic conventions on attributes added by processors, not necessarily by instrumentations? if there is a general agreement from your side on this PR, we can go ahead make the required text changes further. Thank you!